src/rsz/src/Resizer.cc

```diff
@@ -1689,6 +1858,82 @@ int Resizer::resizeToTargetSlew(const Pin* drvr_pin)
   return 0;
 }
 
+bool Resizer::getCin(const LibertyCell* cell, float& cin)
+{
+  sta::LibertyCellPortIterator port_iter(cell);
+  int nports = 0;
+  cin = 0;
+  while (port_iter.hasNext()) {
+    const LibertyPort* port = port_iter.next();
+    if (port->direction() == sta::PortDirection::input()) {
+      cin += port->capacitance();
+      nports++;
+    }
+  }
+  if (nports) {
+    cin /= nports;
+    return true;
+  }
+  return false;
+}
+
+int Resizer::resizeToCapRatio(const Pin* drvr_pin, bool upsize_only)
+{
+  float max_cap_ratio = 4.0f;
+  Instance* inst = network_->instance(drvr_pin);
+  LibertyCell* cell = inst ? network_->libertyCell(inst) : nullptr;
+  if (!network_->isTopLevelPort(drvr_pin) && inst && !dontTouch(inst) && cell
+      && isLogicStdCell(inst)) {
+    float cin, load_cap;
+    ensureWireParasitic(drvr_pin);
+
+    // Includes net parasitic capacitance.
+    load_cap = graph_delay_calc_->loadCap(drvr_pin, tgt_slew_dcalc_ap_);
+    if (load_cap > 0.0 && getCin(cell, cin)) {
+      if (cell->isBuffer()) {
+        max_cap_ratio = 9.0f;
```

Comment:
why?

Reply:
It's mimicking ABC's value. I expect we will want to tune this further (probably as a set_opt_config parameter)